File: cpp/problem002.cpp
#include <iostream>
using namespace std;

// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

int main()
{
    int fib1 = 1;
    int fib2 = 1;
    int sum = 0;
    int result = 0;

    while ( result < 4000000)
    {
        if (result % 2 ==0)
        {
            sum += result;
        }
        
        result = fib1 + fib2;
        fib2 = fib1;
        fib1 = result;
    }
    
    cout << sum << endl;

    return 0;
}

File: python/problem002.py
'''By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.'''

sum = 0
fib1, fib2 = 0, 1
i = 0

while i < 4000000:
    # add even numbers to sum
    if (i % 2 == 0):
        sum += i
    
    # calculate next number in fibonacci sequence
    i = fib1 + fib2
    fib1 = fib2
    fib2 = i

print(sum)